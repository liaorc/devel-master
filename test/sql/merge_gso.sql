--#
--#       Gpu Sort in Merge Join TestCases. 
--#

set pg_strom.debug_force_gpusort to on;
set pg_strom.gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set pg_strom.enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;
set pg_strom.enable_gpuhashjoin to off;
set enable_hashjoin to off;

--smallint
prepare t1 as select * from (select row_number() over (order by t1.smlint_x desc) as rowid,t1.smlint_x,t2.smlint_x from strom_test t1, strom_test t2 where t1.smlint_x=t2.smlint_x and t1.id%100=0) as t where t.rowid%1000=0;
execute t1;
deallocate t1;

prepare t1 as select * from (select row_number() over (order by t1.smlint_x asc) as rowid,t1.smlint_x,t2.smlint_x from strom_test t1, strom_test t2 where t1.smlint_x=t2.smlint_x and t1.id%100=0) as t where t.rowid%1000=0;
execute t1;
deallocate t1;

--integer
prepare t1 as select * from (select row_number() over (order by t1.integer_x desc) as rowid,t1.integer_x,t2.integer_x from strom_test t1, strom_test t2 where t1.integer_x=t2.integer_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

prepare t1 as select * from (select row_number() over (order by t1.integer_x asc) as rowid,t1.integer_x,t2.integer_x from strom_test t1, strom_test t2 where t1.integer_x=t2.integer_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

--bigint
prepare t1 as select * from (select row_number() over (order by t1.bigint_x desc) as rowid,t1.bigint_x,t2.bigint_x from strom_test t1, strom_test t2 where t1.bigint_x=t2.bigint_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

prepare t1 as select * from (select row_number() over (order by t1.bigint_x asc) as rowid,t1.bigint_x,t2.bigint_x from strom_test t1, strom_test t2 where t1.bigint_x=t2.bigint_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

--real
prepare t1 as select * from (select row_number() over (order by t1.real_x desc) as rowid,t1.real_x,t2.real_x from strom_test t1, strom_test t2 where t1.real_x=t2.real_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

prepare t1 as select * from (select row_number() over (order by t1.real_x asc) as rowid,t1.real_x,t2.real_x from strom_test t1, strom_test t2 where t1.real_x=t2.real_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

--float
prepare t1 as select * from (select row_number() over (order by t1.float_x desc) as rowid,t1.float_x,t2.float_x from strom_test t1, strom_test t2 where t1.float_x=t2.float_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

prepare t1 as select * from (select row_number() over (order by t1.float_x asc) as rowid,t1.float_x,t2.float_x from strom_test t1, strom_test t2 where t1.float_x=t2.float_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

--numeric
prepare t1 as select * from (select row_number() over (order by t1.nume_x desc) as rowid,t1.nume_x,t2.nume_x from strom_test t1, strom_test t2 where t1.nume_x=t2.nume_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

prepare t1 as select * from (select row_number() over (order by t1.nume_x asc) as rowid,t1.nume_x,t2.nume_x from strom_test t1, strom_test t2 where t1.nume_x=t2.nume_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
deallocate t1;

