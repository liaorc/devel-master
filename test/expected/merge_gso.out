--#
--#       Gpu Sort in Merge Join TestCases. 
--#
set gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;
set enable_gpuhashjoin to off;
set enable_hashjoin to off;
--smallint
prepare t1 as select * from (select row_number() over (order by t1.smlint_x desc) as rowid,t1.smlint_x,t2.smlint_x from strom_test t1, strom_test t2 where t1.smlint_x=t2.smlint_x and t1.id%100=0) as t where t.rowid%1000=0;
execute t1;
 rowid  | smlint_x | smlint_x 
--------+----------+----------
   1000 |       31 |       31
   2000 |       30 |       30
   3000 |       30 |       30
   4000 |       29 |       29
   5000 |       29 |       29
   6000 |       28 |       28
   7000 |       28 |       28
   8000 |       26 |       26
   9000 |       26 |       26
  10000 |       26 |       26
  11000 |       25 |       25
  12000 |       24 |       24
  13000 |       24 |       24
  14000 |       23 |       23
  15000 |       22 |       22
  16000 |       22 |       22
  17000 |       22 |       22
  18000 |       22 |       22
  19000 |       21 |       21
  20000 |       21 |       21
  21000 |       21 |       21
  22000 |       20 |       20
  23000 |       19 |       19
  24000 |       19 |       19
  25000 |       18 |       18
  26000 |       18 |       18
  27000 |       17 |       17
  28000 |       16 |       16
  29000 |       16 |       16
  30000 |       16 |       16
  31000 |       15 |       15
  32000 |       14 |       14
  33000 |       14 |       14
  34000 |       13 |       13
  35000 |       12 |       12
  36000 |       12 |       12
  37000 |       11 |       11
  38000 |       11 |       11
  39000 |       10 |       10
  40000 |        9 |        9
  41000 |        9 |        9
  42000 |        9 |        9
  43000 |        8 |        8
  44000 |        7 |        7
  45000 |        6 |        6
  46000 |        6 |        6
  47000 |        4 |        4
  48000 |        4 |        4
  49000 |        4 |        4
  50000 |        4 |        4
  51000 |        3 |        3
  52000 |        2 |        2
  53000 |        2 |        2
  54000 |        1 |        1
  55000 |        1 |        1
  56000 |        1 |        1
  57000 |        0 |        0
  58000 |       -1 |       -1
  59000 |       -1 |       -1
  60000 |       -2 |       -2
  61000 |       -2 |       -2
  62000 |       -2 |       -2
  63000 |       -2 |       -2
  64000 |       -3 |       -3
  65000 |       -3 |       -3
  66000 |       -4 |       -4
  67000 |       -4 |       -4
  68000 |       -5 |       -5
  69000 |       -5 |       -5
  70000 |       -6 |       -6
  71000 |       -6 |       -6
  72000 |       -7 |       -7
  73000 |       -8 |       -8
  74000 |       -8 |       -8
  75000 |       -9 |       -9
  76000 |      -10 |      -10
  77000 |      -10 |      -10
  78000 |      -10 |      -10
  79000 |      -10 |      -10
  80000 |      -11 |      -11
  81000 |      -11 |      -11
  82000 |      -12 |      -12
  83000 |      -12 |      -12
  84000 |      -12 |      -12
  85000 |      -13 |      -13
  86000 |      -14 |      -14
  87000 |      -14 |      -14
  88000 |      -14 |      -14
  89000 |      -14 |      -14
  90000 |      -15 |      -15
  91000 |      -15 |      -15
  92000 |      -16 |      -16
  93000 |      -16 |      -16
  94000 |      -17 |      -17
  95000 |      -17 |      -17
  96000 |      -18 |      -18
  97000 |      -18 |      -18
  98000 |      -19 |      -19
  99000 |      -19 |      -19
 100000 |      -19 |      -19
 101000 |      -20 |      -20
 102000 |      -21 |      -21
 103000 |      -22 |      -22
 104000 |      -22 |      -22
 105000 |      -23 |      -23
 106000 |      -24 |      -24
 107000 |      -25 |      -25
 108000 |      -25 |      -25
 109000 |      -26 |      -26
 110000 |      -26 |      -26
 111000 |      -27 |      -27
 112000 |      -27 |      -27
 113000 |      -27 |      -27
 114000 |      -28 |      -28
 115000 |      -28 |      -28
 116000 |      -28 |      -28
 117000 |      -28 |      -28
 118000 |      -29 |      -29
 119000 |      -30 |      -30
 120000 |      -30 |      -30
 121000 |      -32 |      -32
 122000 |      -32 |      -32
(122 rows)

deallocate t1;
prepare t1 as select * from (select row_number() over (order by t1.smlint_x asc) as rowid,t1.smlint_x,t2.smlint_x from strom_test t1, strom_test t2 where t1.smlint_x=t2.smlint_x and t1.id%100=0) as t where t.rowid%1000=0;
execute t1;
 rowid  | smlint_x | smlint_x 
--------+----------+----------
   1000 |      -32 |      -32
   2000 |      -30 |      -30
   3000 |      -30 |      -30
   4000 |      -30 |      -30
   5000 |      -28 |      -28
   6000 |      -28 |      -28
   7000 |      -28 |      -28
   8000 |      -28 |      -28
   9000 |      -27 |      -27
  10000 |      -27 |      -27
  11000 |      -27 |      -27
  12000 |      -27 |      -27
  13000 |      -26 |      -26
  14000 |      -26 |      -26
  15000 |      -25 |      -25
  16000 |      -24 |      -24
  17000 |      -23 |      -23
  18000 |      -23 |      -23
  19000 |      -22 |      -22
  20000 |      -21 |      -21
  21000 |      -20 |      -20
  22000 |      -19 |      -19
  23000 |      -19 |      -19
  24000 |      -19 |      -19
  25000 |      -18 |      -18
  26000 |      -18 |      -18
  27000 |      -17 |      -17
  28000 |      -17 |      -17
  29000 |      -16 |      -16
  30000 |      -16 |      -16
  31000 |      -15 |      -15
  32000 |      -15 |      -15
  33000 |      -14 |      -14
  34000 |      -14 |      -14
  35000 |      -14 |      -14
  36000 |      -14 |      -14
  37000 |      -13 |      -13
  38000 |      -12 |      -12
  39000 |      -12 |      -12
  40000 |      -12 |      -12
  41000 |      -11 |      -11
  42000 |      -11 |      -11
  43000 |      -10 |      -10
  44000 |      -10 |      -10
  45000 |      -10 |      -10
  46000 |      -10 |      -10
  47000 |       -9 |       -9
  48000 |       -8 |       -8
  49000 |       -8 |       -8
  50000 |       -7 |       -7
  51000 |       -6 |       -6
  52000 |       -6 |       -6
  53000 |       -6 |       -6
  54000 |       -5 |       -5
  55000 |       -4 |       -4
  56000 |       -4 |       -4
  57000 |       -3 |       -3
  58000 |       -3 |       -3
  59000 |       -3 |       -3
  60000 |       -2 |       -2
  61000 |       -2 |       -2
  62000 |       -2 |       -2
  63000 |       -1 |       -1
  64000 |       -1 |       -1
  65000 |        0 |        0
  66000 |        1 |        1
  67000 |        1 |        1
  68000 |        1 |        1
  69000 |        2 |        2
  70000 |        2 |        2
  71000 |        3 |        3
  72000 |        4 |        4
  73000 |        4 |        4
  74000 |        4 |        4
  75000 |        4 |        4
  76000 |        6 |        6
  77000 |        6 |        6
  78000 |        7 |        7
  79000 |        8 |        8
  80000 |        9 |        9
  81000 |        9 |        9
  82000 |        9 |        9
  83000 |       10 |       10
  84000 |       11 |       11
  85000 |       11 |       11
  86000 |       12 |       12
  87000 |       12 |       12
  88000 |       13 |       13
  89000 |       14 |       14
  90000 |       14 |       14
  91000 |       15 |       15
  92000 |       16 |       16
  93000 |       16 |       16
  94000 |       16 |       16
  95000 |       17 |       17
  96000 |       18 |       18
  97000 |       18 |       18
  98000 |       18 |       18
  99000 |       19 |       19
 100000 |       20 |       20
 101000 |       21 |       21
 102000 |       21 |       21
 103000 |       21 |       21
 104000 |       21 |       21
 105000 |       22 |       22
 106000 |       22 |       22
 107000 |       22 |       22
 108000 |       23 |       23
 109000 |       24 |       24
 110000 |       24 |       24
 111000 |       24 |       24
 112000 |       25 |       25
 113000 |       26 |       26
 114000 |       26 |       26
 115000 |       27 |       27
 116000 |       28 |       28
 117000 |       29 |       29
 118000 |       29 |       29
 119000 |       29 |       29
 120000 |       30 |       30
 121000 |       31 |       31
 122000 |       32 |       32
(122 rows)

deallocate t1;
--integer
prepare t1 as select * from (select row_number() over (order by t1.integer_x desc) as rowid,t1.integer_x,t2.integer_x from strom_test t1, strom_test t2 where t1.integer_x=t2.integer_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid | integer_x | integer_x 
-------+-----------+-----------
    10 |   1998626 |   1998626
    20 |   1914986 |   1914986
    30 |   1752987 |   1752987
    40 |   1633749 |   1633749
    50 |   1417274 |   1417274
    60 |   1273818 |   1273818
    70 |   1083772 |   1083772
    80 |    921409 |    921409
    90 |    874668 |    874668
   100 |    804859 |    804859
   110 |    574288 |    574288
   120 |    374638 |    374638
   130 |    281936 |    281936
   140 |    183996 |    183996
   150 |     33992 |     33992
   160 |   -139442 |   -139442
   170 |   -316384 |   -316384
   180 |   -510199 |   -510199
   190 |   -568631 |   -568631
   200 |   -783509 |   -783509
   210 |   -949668 |   -949668
   220 |  -1140932 |  -1140932
   230 |  -1260720 |  -1260720
   240 |  -1413153 |  -1413153
   250 |  -1518441 |  -1518441
   260 |  -1727189 |  -1727189
   270 |  -1856438 |  -1856438
   280 |  -1954772 |  -1954772
   290 |  -2140602 |  -2140602
(29 rows)

deallocate t1;
prepare t1 as select * from (select row_number() over (order by t1.integer_x asc) as rowid,t1.integer_x,t2.integer_x from strom_test t1, strom_test t2 where t1.integer_x=t2.integer_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid | integer_x | integer_x 
-------+-----------+-----------
    10 |  -1967031 |  -1967031
    20 |  -1856856 |  -1856856
    30 |  -1730920 |  -1730920
    40 |  -1527276 |  -1527276
    50 |  -1425910 |  -1425910
    60 |  -1316903 |  -1316903
    70 |  -1141242 |  -1141242
    80 |   -987445 |   -987445
    90 |   -795105 |   -795105
   100 |   -592485 |   -592485
   110 |   -514868 |   -514868
   120 |   -321596 |   -321596
   130 |   -153844 |   -153844
   140 |     -1553 |     -1553
   150 |    177423 |    177423
   160 |    273507 |    273507
   170 |    368386 |    368386
   180 |    571052 |    571052
   190 |    800891 |    800891
   200 |    874224 |    874224
   210 |    918442 |    918442
   220 |   1071382 |   1071382
   230 |   1265004 |   1265004
   240 |   1410919 |   1410919
   250 |   1626822 |   1626822
   260 |   1718563 |   1718563
   270 |   1897091 |   1897091
   280 |   1985563 |   1985563
   290 |   2095362 |   2095362
(29 rows)

deallocate t1;
--bigint
prepare t1 as select * from (select row_number() over (order by t1.bigint_x desc) as rowid,t1.bigint_x,t2.bigint_x from strom_test t1, strom_test t2 where t1.bigint_x=t2.bigint_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid |     bigint_x      |     bigint_x      
-------+-------------------+-------------------
    10 |  8594477984499368 |  8594477984499368
    20 |  7923571640091279 |  7923571640091279
    30 |  7212538567287898 |  7212538567287898
    40 |  6916096338220286 |  6916096338220286
    50 |  6205740779327980 |  6205740779327980
    60 |  5744271763420742 |  5744271763420742
    70 |  5145890047794348 |  5145890047794348
    80 |  4803860937062744 |  4803860937062744
    90 |  4362544388744676 |  4362544388744676
   100 |  3392359573485519 |  3392359573485519
   110 |  2760038154407772 |  2760038154407772
   120 |  1709381264227697 |  1709381264227697
   130 |   912993649220911 |   912993649220911
   140 |   550322955729502 |   550322955729502
   150 |  -331913629525869 |  -331913629525869
   160 | -1144090201442746 | -1144090201442746
   170 | -1918337659470610 | -1918337659470610
   180 | -2300353551383134 | -2300353551383134
   190 | -2932895224973754 | -2932895224973754
   200 | -3536550597798920 | -3536550597798920
   210 | -4143317807574024 | -4143317807574024
   220 | -5245604606993498 | -5245604606993498
   230 | -5791544527122596 | -5791544527122596
   240 | -6250665703452443 | -6250665703452443
   250 | -6669085295784231 | -6669085295784231
   260 | -7343900807755989 | -7343900807755989
   270 | -8144051720352170 | -8144051720352170
   280 | -8779678997232484 | -8779678997232484
(28 rows)

deallocate t1;
prepare t1 as select * from (select row_number() over (order by t1.bigint_x asc) as rowid,t1.bigint_x,t2.bigint_x from strom_test t1, strom_test t2 where t1.bigint_x=t2.bigint_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid |     bigint_x      |     bigint_x      
-------+-------------------+-------------------
    10 | -8511719994839532 | -8511719994839532
    20 | -7967589183654986 | -7967589183654986
    30 | -7070177894827295 | -7070177894827295
    40 | -6586104624954999 | -6586104624954999
    50 | -6189261032762900 | -6189261032762900
    60 | -5533208978776916 | -5533208978776916
    70 | -5099055533965443 | -5099055533965443
    80 | -4016772695050944 | -4016772695050944
    90 | -3513375924932313 | -3513375924932313
   100 | -2796074957245776 | -2796074957245776
   110 | -2254717758631576 | -2254717758631576
   120 | -1869858430865375 | -1869858430865375
   130 | -1049891415338779 | -1049891415338779
   140 |  -165055438566457 |  -165055438566457
   150 |   614317049316901 |   614317049316901
   160 |  1065436203358093 |  1065436203358093
   170 |  1813882757751767 |  1813882757751767
   180 |  2820054601472934 |  2820054601472934
   190 |  3740979358777475 |  3740979358777475
   200 |  4526675438679884 |  4526675438679884
   210 |  4873511270650741 |  4873511270650741
   220 |  5344877265597497 |  5344877265597497
   230 |  5862563782751945 |  5862563782751945
   240 |  6357462848406290 |  6357462848406290
   250 |  6963228703736201 |  6963228703736201
   260 |  7357622390748086 |  7357622390748086
   270 |  8101327438685929 |  8101327438685929
   280 |  8778598838727344 |  8778598838727344
(28 rows)

deallocate t1;
--real
prepare t1 as select * from (select row_number() over (order by t1.real_x desc) as rowid,t1.real_x,t2.real_x from strom_test t1, strom_test t2 where t1.real_x=t2.real_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid | real_x  | real_x  
-------+---------+---------
    10 |  0.9516 |  0.9516
    20 |  0.9292 |  0.9292
    30 |  0.8931 |  0.8931
    40 |   0.886 |   0.886
    50 |  0.8517 |  0.8517
    60 |  0.8115 |  0.8115
    70 |  0.7735 |  0.7735
    80 |  0.7535 |  0.7535
    90 |  0.7145 |  0.7145
   100 |  0.6799 |  0.6799
   110 |    0.67 |    0.67
   120 |  0.6469 |  0.6469
   130 |  0.6251 |  0.6251
   140 |  0.5746 |  0.5746
   150 |  0.5369 |  0.5369
   160 |  0.5117 |  0.5117
   170 |   0.487 |   0.487
   180 |   0.473 |   0.473
   190 |  0.4433 |  0.4433
   200 |  0.4007 |  0.4007
   210 |  0.3848 |  0.3848
   220 |  0.3315 |  0.3315
   230 |  0.3179 |  0.3179
   240 |  0.2633 |  0.2633
   250 |  0.2296 |  0.2296
   260 |  0.1924 |  0.1924
   270 |  0.1705 |  0.1705
   280 |  0.1362 |  0.1362
   290 |  0.1074 |  0.1074
   300 |  0.0829 |  0.0829
   310 |  0.0569 |  0.0569
   320 |   0.024 |   0.024
   330 |  0.0097 |  0.0097
   340 | -0.0137 | -0.0137
   350 | -0.0346 | -0.0346
   360 | -0.0751 | -0.0751
   370 |  -0.088 |  -0.088
   380 | -0.1003 | -0.1003
   390 | -0.1616 | -0.1616
   400 | -0.2279 | -0.2279
   410 | -0.2825 | -0.2825
   420 | -0.3125 | -0.3125
   430 | -0.3461 | -0.3461
   440 | -0.3613 | -0.3613
   450 | -0.3903 | -0.3903
   460 |  -0.424 |  -0.424
   470 | -0.4502 | -0.4502
   480 | -0.5397 | -0.5397
   490 | -0.5677 | -0.5677
   500 | -0.5801 | -0.5801
   510 | -0.6189 | -0.6189
   520 | -0.6513 | -0.6513
   530 | -0.6649 | -0.6649
   540 | -0.6992 | -0.6992
   550 | -0.7266 | -0.7266
   560 |  -0.733 |  -0.733
   570 | -0.7523 | -0.7523
   580 | -0.7746 | -0.7746
   590 | -0.8153 | -0.8153
   600 | -0.8372 | -0.8372
   610 | -0.8555 | -0.8555
   620 | -0.8865 | -0.8865
   630 | -0.8865 | -0.8865
   640 | -0.9095 | -0.9095
   650 | -0.9283 | -0.9283
   660 |  -0.945 |  -0.945
   670 | -0.9745 | -0.9745
(67 rows)

deallocate t1;
prepare t1 as select * from (select row_number() over (order by t1.real_x asc) as rowid,t1.real_x,t2.real_x from strom_test t1, strom_test t2 where t1.real_x=t2.real_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid | real_x  | real_x  
-------+---------+---------
    10 | -0.9745 | -0.9745
    20 |  -0.945 |  -0.945
    30 | -0.9203 | -0.9203
    40 | -0.9006 | -0.9006
    50 | -0.8865 | -0.8865
    60 | -0.8865 | -0.8865
    70 | -0.8555 | -0.8555
    80 | -0.8336 | -0.8336
    90 | -0.8013 | -0.8013
   100 | -0.7746 | -0.7746
   110 | -0.7523 | -0.7523
   120 |  -0.733 |  -0.733
   130 | -0.7199 | -0.7199
   140 | -0.6992 | -0.6992
   150 | -0.6613 | -0.6613
   160 | -0.6513 | -0.6513
   170 | -0.6153 | -0.6153
   180 | -0.5801 | -0.5801
   190 | -0.5612 | -0.5612
   200 | -0.5397 | -0.5397
   210 | -0.4502 | -0.4502
   220 |  -0.424 |  -0.424
   230 | -0.3903 | -0.3903
   240 | -0.3592 | -0.3592
   250 | -0.3461 | -0.3461
   260 | -0.3054 | -0.3054
   270 | -0.2825 | -0.2825
   280 | -0.2225 | -0.2225
   290 |  -0.158 |  -0.158
   300 | -0.1003 | -0.1003
   310 |  -0.088 |  -0.088
   320 | -0.0689 | -0.0689
   330 | -0.0346 | -0.0346
   340 | -0.0008 | -0.0008
   350 |   0.013 |   0.013
   360 |  0.0278 |  0.0278
   370 |  0.0659 |  0.0659
   380 |  0.0829 |  0.0829
   390 |  0.1074 |  0.1074
   400 |   0.143 |   0.143
   410 |  0.1705 |  0.1705
   420 |  0.1924 |  0.1924
   430 |   0.231 |   0.231
   440 |  0.2633 |  0.2633
   450 |  0.3201 |  0.3201
   460 |  0.3396 |  0.3396
   470 |  0.3876 |  0.3876
   480 |  0.4055 |  0.4055
   490 |  0.4433 |  0.4433
   500 |  0.4751 |  0.4751
   510 |   0.487 |   0.487
   520 |  0.5157 |  0.5157
   530 |  0.5424 |  0.5424
   540 |  0.5746 |  0.5746
   550 |  0.6251 |  0.6251
   560 |  0.6469 |  0.6469
   570 |  0.6703 |  0.6703
   580 |  0.6799 |  0.6799
   590 |   0.722 |   0.722
   600 |  0.7535 |  0.7535
   610 |  0.7786 |  0.7786
   620 |  0.8392 |  0.8392
   630 |   0.858 |   0.858
   640 |   0.886 |   0.886
   650 |  0.8931 |  0.8931
   660 |  0.9292 |  0.9292
   670 |  0.9628 |  0.9628
(67 rows)

deallocate t1;
--float
prepare t1 as select * from (select row_number() over (order by t1.float_x desc) as rowid,t1.float_x,t2.float_x from strom_test t1, strom_test t2 where t1.float_x=t2.float_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid |     float_x      |     float_x      
-------+------------------+------------------
    10 |  0.9166077389382 |  0.9166077389382
    20 |  0.8256782633252 |  0.8256782633252
    30 |  0.7711737770587 |  0.7711737770587
    40 |   0.742534075398 |   0.742534075398
    50 |  0.6947398427874 |  0.6947398427874
    60 |  0.6134519930929 |  0.6134519930929
    70 |  0.4664505515248 |  0.4664505515248
    80 |  0.4118170444854 |  0.4118170444854
    90 |  0.3621213920414 |  0.3621213920414
   100 |   0.325676355511 |   0.325676355511
   110 |    0.21904671425 |    0.21904671425
   120 |    0.10670165997 |    0.10670165997
   130 |  0.0542140416801 |  0.0542140416801
   140 |   0.005014274735 |   0.005014274735
   150 | -0.0661425888538 | -0.0661425888538
   160 |  -0.104260857217 |  -0.104260857217
   170 | -0.1537507371977 | -0.1537507371977
   180 | -0.2565650912002 | -0.2565650912002
   190 | -0.3295058198273 | -0.3295058198273
   200 | -0.4061025194824 | -0.4061025194824
   210 | -0.4960924419574 | -0.4960924419574
   220 | -0.5465021757409 | -0.5465021757409
   230 | -0.6195399849676 | -0.6195399849676
   240 | -0.6854076851159 | -0.6854076851159
   250 | -0.7376694818959 | -0.7376694818959
   260 | -0.8118678648025 | -0.8118678648025
   270 | -0.8899431545287 | -0.8899431545287
   280 | -0.9483382562175 | -0.9483382562175
(28 rows)

deallocate t1;
prepare t1 as select * from (select row_number() over (order by t1.float_x asc) as rowid,t1.float_x,t2.float_x from strom_test t1, strom_test t2 where t1.float_x=t2.float_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid |     float_x      |     float_x      
-------+------------------+------------------
    10 | -0.9483382562175 | -0.9483382562175
    20 | -0.8899431545287 | -0.8899431545287
    30 | -0.8118678648025 | -0.8118678648025
    40 | -0.7376694818959 | -0.7376694818959
    50 | -0.6854076851159 | -0.6854076851159
    60 | -0.6195399849676 | -0.6195399849676
    70 | -0.5465021757409 | -0.5465021757409
    80 | -0.4960924419574 | -0.4960924419574
    90 | -0.4061025194824 | -0.4061025194824
   100 | -0.3295058198273 | -0.3295058198273
   110 | -0.2565650912002 | -0.2565650912002
   120 | -0.1537507371977 | -0.1537507371977
   130 |  -0.104260857217 |  -0.104260857217
   140 | -0.0661425888538 | -0.0661425888538
   150 |   0.005014274735 |   0.005014274735
   160 |  0.0542140416801 |  0.0542140416801
   170 |    0.10670165997 |    0.10670165997
   180 |    0.21904671425 |    0.21904671425
   190 |   0.325676355511 |   0.325676355511
   200 |  0.3621213920414 |  0.3621213920414
   210 |  0.4118170444854 |  0.4118170444854
   220 |  0.4664505515248 |  0.4664505515248
   230 |  0.6134519930929 |  0.6134519930929
   240 |  0.6947398427874 |  0.6947398427874
   250 |   0.742534075398 |   0.742534075398
   260 |  0.7711737770587 |  0.7711737770587
   270 |  0.8256782633252 |  0.8256782633252
   280 |  0.9166077389382 |  0.9166077389382
(28 rows)

deallocate t1;
--numeric
prepare t1 as select * from (select row_number() over (order by t1.nume_x desc) as rowid,t1.nume_x,t2.nume_x from strom_test t1, strom_test t2 where t1.nume_x=t2.nume_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid |       nume_x        |       nume_x        
-------+---------------------+---------------------
    10 |   0.915784781333059 |   0.915784781333059
    20 |   0.844137607142329 |   0.844137607142329
    30 |   0.786268042866141 |   0.786268042866141
    40 |    0.68250407744199 |    0.68250407744199
    50 |   0.629790987819433 |   0.629790987819433
    60 |   0.550148699432611 |   0.550148699432611
    70 |   0.472813767381012 |   0.472813767381012
    80 |    0.39022592548281 |    0.39022592548281
    90 |   0.326096843462437 |   0.326096843462437
   100 |   0.279945631045848 |   0.279945631045848
   110 |   0.207269317470491 |   0.207269317470491
   120 |   0.138154617510736 |   0.138154617510736
   130 |  0.0786340609192848 |  0.0786340609192848
   140 |  0.0330492472276092 |  0.0330492472276092
   150 | -0.0268530733883381 | -0.0268530733883381
   160 | -0.0666161319240928 | -0.0666161319240928
   170 |  -0.139068128541112 |  -0.139068128541112
   180 |  -0.224808412604034 |  -0.224808412604034
   190 |  -0.300269112922251 |  -0.300269112922251
   200 |  -0.359443863853812 |  -0.359443863853812
   210 |  -0.457400043960661 |  -0.457400043960661
   220 |  -0.506501241587102 |  -0.506501241587102
   230 |   -0.55728673376143 |   -0.55728673376143
   240 |   -0.64384410623461 |   -0.64384410623461
   250 |  -0.713439894840121 |  -0.713439894840121
   260 |  -0.803012616932392 |  -0.803012616932392
   270 |  -0.860798018518835 |  -0.860798018518835
   280 |   -0.93033471563831 |   -0.93033471563831
(28 rows)

deallocate t1;
prepare t1 as select * from (select row_number() over (order by t1.nume_x asc) as rowid,t1.nume_x,t2.nume_x from strom_test t1, strom_test t2 where t1.nume_x=t2.nume_x and t1.id%100=0) as t where t.rowid%10=0;
execute t1;
 rowid |       nume_x        |       nume_x        
-------+---------------------+---------------------
    10 |  -0.925957737490535 |  -0.925957737490535
    20 |  -0.841332031413913 |  -0.841332031413913
    30 |  -0.798875930719078 |  -0.798875930719078
    40 |   -0.70831117965281 |   -0.70831117965281
    50 |  -0.635598768480122 |  -0.635598768480122
    60 |  -0.545667663682252 |  -0.545667663682252
    70 |  -0.506388329900801 |  -0.506388329900801
    80 |  -0.451121483463794 |  -0.451121483463794
    90 |  -0.358608504291624 |  -0.358608504291624
   100 |  -0.293440037872642 |  -0.293440037872642
   110 |  -0.216976442374289 |  -0.216976442374289
   120 |  -0.126575714442879 |  -0.126575714442879
   130 | -0.0642513819038868 | -0.0642513819038868
   140 | -0.0265072244219482 | -0.0265072244219482
   150 |   0.033942733425647 |   0.033942733425647
   160 |  0.0793187296949327 |  0.0793187296949327
   170 |   0.139033701270819 |   0.139033701270819
   180 |   0.212715436238796 |   0.212715436238796
   190 |   0.282154576852918 |   0.282154576852918
   200 |   0.338509722147137 |   0.338509722147137
   210 |   0.392388930078596 |   0.392388930078596
   220 |   0.481465172022581 |   0.481465172022581
   230 |   0.552436067722738 |   0.552436067722738
   240 |     0.6313068382442 |     0.6313068382442
   250 |   0.684447686187923 |   0.684447686187923
   260 |   0.801311953924596 |   0.801311953924596
   270 |   0.855744058731943 |   0.855744058731943
   280 |    0.92457263963297 |    0.92457263963297
(28 rows)

deallocate t1;
